modules/service/src/main/java11/org/mule/runtime/module/service/api/artifact/ServiceModuleLayerFactory.java

```diff
@@ -66,26 +78,45 @@ private MuleServiceClassLoader(String artifactId, ArtifactDescriptor artifactDes
   @Deprecated
   public ArtifactClassLoader create(String artifactId, ServiceDescriptor descriptor, ClassLoader parent,
                                     ClassLoaderLookupPolicy lookupPolicy) {
-    // TODO TD-0144818 remove this special case
-    if (artifactId.equals("service/DataWeave service")
-        && !getBoolean(getProperty(CLASSLOADER_SERVICE_JPMS_MODULE_LAYER_DATAWEAVE, "false"))) {
-      return new MuleArtifactClassLoader(artifactId, descriptor, descriptor.getClassLoaderConfiguration().getUrls(), parent,
-                                         lookupPolicy);
+    URL[] classLoaderConfigurationUrls = descriptor.getClassLoaderConfiguration().getUrls();
+
+    if (artifactId.equals("service/DataWeave service")) {
+      // TODO TD-0144818 remove this special case
+      if (!getBoolean(CLASSLOADER_SERVICE_JPMS_MODULE_LAYER_DATAWEAVE)) {
```

Comment:
In what case would this property be used?

Reply:
for DW team to be able to test their stuff as they work on it